src/utilities.jl

```diff
@@ -145,3 +145,71 @@ end
 
 check_positive_definite(M::AbstractMatrix; kwargs...) =
   check_positive_definite(LinearOperator(M); kwargs...)
+
+
+  """
+  solve_shifted_system!(B, 
+                        z, 
+                        σ,
+                        x, 
+                        )
+
+Solves linear system (B + σI) x = z, where B is a forward L-BFGS operator and σ ≥ 0.
+
+### Parameters
+
+- `B::LBFGSOperator{T,I,F1,F2,F3}`: forward L-BFGS operator, a LinearOperator that models a matrix of dimension nxn;
+- `z::AbstractVector{T}`: The vector of length n.
+- `σ::T`: Nonnegative shift scalar.
+- `x::AbstractVector{T}`: A preallocated vector a vector of length n that is used to store the solution x.
+
+### Returns
+
+- `x::AbstractVector{T}`: The solution vector `x` size n, the solution to `(B + σI) x = b`.
+
+### Method
+
+The method uses a two-loop recursion-like approach with modifications to handle the shift `σ`.
+
+### References
+Erway, J. B., Jain, V., & Marcia, R. F. (2014). Shifted L-BFGS Systems. Optimization Methods and Software, 29(5), 992-1004.
+"""
+
+function solve_shifted_system!(
+  B::LBFGSOperator{T, I, F1, F2, F3},
+  z::AbstractVector{T},
+  σ::T,
```

Comment:
We need to check that `σ ≥ 0`. The procedure doesn’t work for `σ < 0`, does it?

You could also implement `ldiv!()` so that it matches the method from LinearAlgebra and calls `solve_shifted_system!()` with `σ = 0`. In the unit tests, you can check that the solve with B matches an operator-vector product with H (the corresponding InverseLBFGSOperator).